[WIP] Make a system for keeping docs and file names in sync

mattcieslak opened the pull request:
The recon scalars are a special case in that they're shared across nodes and we have a good system for documenting them. But the other outputs from each node are currently just being sent to regular datasinks with the bids entities hard coded into the workflow code.

This PR sets up a new directory of yaml files for nonscalar outputs and their bids entities / metadata. Since BEP016 doesn't cover these, there will likely be significant changes in the file names, so having something flexible is going to be great. 

Comments would be appreciated!

mattcieslak committed:
minimal setup

mattcieslak committed:
minimal setup

tsalo commented on qsirecon/workflows/recon/amico.py:
```diff
@@ -141,12 +143,12 @@ def init_amico_noddi_fit_wf(
         ])  # fmt:skip
 
     if qsirecon_suffix:
+        derivatives_config = load_yaml(load_data("scalars/amico_noddi.yaml"))
```
I think this should be `nonscalars`

mattcieslak committed:
use real file

mattcieslak commented:
Add paths to yaml files to docstrings of workflows

tsalo commented:
It looks like the problem is that you're trying to resolve #78 here, but you need to change the io_spec.json file to make that work.

mattcieslak committed:
io_spec

mattcieslak committed:
one more io_spec

mattcieslak committed:
Update expected files

mattcieslak committed:
Update expected files

tsalo committed:
Remove unused `--longitudinal` argument (#88)

Remove unused `--longitudinal` argument.

mattcieslak committed:
repeat

mattcieslak commented:
Alright, this looks pretty good in the docs. I'm going to update it for the rest of the workflows too

cookpa committed:
Use file lock to avoid concurrent edits to dataset_description.json (#91)

* ENH: use file lock to avoid concurrent edits to dataset_description.json

BUG: docker tag in dataset_description was for fmriprep

* ENH: Move lock outside the check for file existence

Ensures only the first process writes the file

* STYLE: fix style issues

ENH: Add file lock for the bids ignore file

* BUG: docker address is pennlinc not pennbbl

tsalo committed:
Make all params and models lower-case (#90)

* Make all params lower-case.

* Make models lower-case too.

* Fix things up.

* Fix underscores in entities.

* fix

* Revert some changes.

* Update pyafq_recon_external_trk_outputs.txt

mattcieslak committed:
temp

mattcieslak committed:
outputs

mattcieslak committed:
Document all non-scalar outputs

mattcieslak committed:
Merge branch 'main' into doc/other_derivatives

mattcieslak committed:
lint
